control-plane-operator/featuregates/featuregates.go

```diff
@@ -0,0 +1,47 @@
+package featuregates
+
+import (
+	"github.com/openshift/hypershift/pkg/featuregates"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"k8s.io/component-base/featuregate"
+)
+
+// Define new featuregates here
+const (
+	Foo featuregate.Feature = "Foo"
+	Bar featuregate.Feature = "Bar"
+)
+
+var (
+	allFeatures = featuregates.NewFeatureSetAwareFeatures()
```

Comment:
I'm fine keeping this source for set<-> features separate from the HO as you proposed so feel free to ignore my comment on the sync discussion